### ISSUE TITLE
Unable to successfully decrypt fields

### ISSUE THREAD

**Issue by MLBZ521** [dmaasland/jamf_decrypt#2]
Hello,

I've been trying to get `jamf_decrypt.py` run successfully the last couple days.

I've tried with the latest "update" for Python3, but the `MySQLdb` library looks like it's no longer supported and I can't get it to install -- not even with Python2.

I've converted the code to use the (official Oracle) `mysql-connector-python` library, which I use in another script that connects to the Jamf Pro database and I've got it working in `jamf_decrypt.py`, however, the decryption part seems to be where things are failing.

I may be missing _something_...

Here's where I'm currently at (with a little bit extra verbosity printed out):

```python
$ python AxM/decrypt_jps_db.py 
SSH Connected!
Connecting to the jamfsoftware database on host 127.0.0.1...Connected!
[+] Got encryption_key:  'bytearray(b'<redacted>')'
[+] key:  'b'<redacted>''
[+] iv:  'b'<redacted>''
[+] Got session_key:  'b'<redacted>''
Connecting to the jamfsoftware database on host 127.0.0.1...Connected!
[-] Found encrypted data in table 'adcs_settings', decrypting...
column:  cat_client_cert_password_encrypted
[+] cipher_text:  'b'<redacted>''
Traceback (most recent call last):
  File "/Users/<user>/code/test/decrypt_jps_db.py", line 534, in <module>
    main()
  File "/Users/<user>/code/test/decrypt_jps_db.py", line 530, in main
    dump_data(jps, jamf_db_cursor)
  File "/Users/<user>/code/test/decrypt_jps_db.py", line 417, in dump_data
    f"\t\t\t\t\t<td>{jps.decrypt(base64.b64decode(row[column_index]))}</td>\n")
  File "/Users/<user>/code/test/decrypt_jps_db.py", line 162, in decrypt
    key, iv = generator.generate_derived_parameters(password, self.salt, self.iterations)
  File "/Users/<user>/code/test/.venv/lib/python3.10/site-packages/jasypt4py/generator.py", line 91, in generate_derived_parameters
    password_bytes = PKCS12ParameterGenerator.pkcs12_password_to_bytes(password)
  File "/Users/<user>/code/test/.venv/lib/python3.10/site-packages/jasypt4py/generator.py", line 45, in pkcs12_password_to_bytes
    digit = ord(password[i])
TypeError: ord() expected string of length 1, but int found
```

Basically, I'm connecting to my server over SSH, connecting to the MySQL database, and then the `get_session_key` function runs and I get a value (not sure if there's a way to validate this value is correct?) and finally, the `dump_data` function runs, which starts through through the tables, and on the first encrypted field (column) the above^^ happens.

Any help or insight would be greatly appreciated!

**Comment by MLBZ521** [comment]
Meh, I almost had it last night.  This morning, I checked the library being called and noticed the value type it was expecting was not matching the value type that was being passed and after checking, I saw what I had done that caused it to fail there.

That said, there were some additional tweaks that I had to make for fields used in my database to not throw exceptions.  I'll clean up my code and make a PR.

Thanks for all the leg work here!
